File: src/main/java/net/flyingfishflash/loremlist/domain/lrmlist/ListNotFoundException.java
package net.flyingfishflash.loremlist.domain.lrmlist;

import java.util.Set;
import java.util.UUID;
import net.flyingfishflash.loremlist.domain.LrmComponentType;
import net.flyingfishflash.loremlist.domain.exceptions.EntityNotFoundException;

public class ListNotFoundException extends EntityNotFoundException {

  public ListNotFoundException(Set<UUID> idCollection, String message) {
    super(idCollection, message, LrmComponentType.List);
  }

  public ListNotFoundException(Set<UUID> idCollection) {
    this(idCollection, null);
  }

  public ListNotFoundException(UUID id, String message) {
    this(Set.of(id), message);
  }

  public ListNotFoundException(UUID id) {
    this(Set.of(id), null);
  }

  public ListNotFoundException() {
    this(Set.of(), null);
  }
}

File: src/main/kotlin/net/flyingfishflash/loremlist/domain/lrmlist/ListNotFoundException.kt
package net.flyingfishflash.loremlist.domain.lrmlist

import net.flyingfishflash.loremlist.domain.LrmComponentType
import net.flyingfishflash.loremlist.domain.exceptions.EntityNotFoundException
import java.util.UUID

class ListNotFoundException(idCollection: Set<UUID> = emptySet(), message: String? = null) :
  EntityNotFoundException(
    idCollection = idCollection,
    message = message,
    lrmComponentType = LrmComponentType.List,
  ) {
  constructor(id: UUID, message: String? = null) : this(setOf(id), message = message)
}
